### PR TITLE
refactor: Add `message` container to `payload`

### PR DESCRIPTION
**Pull request by fabriziodemaria**
Also minor changes in the various interfaces and how events are stored on disk

### CODE REVIEW

File: Provider/src/main/java/com/spotify/confidence/EventSenderUploader.kt [hunk 1]
```diff
@@ -52,11 +52,13 @@ internal class EventSenderUploaderImpl(
     }
 
     override suspend fun upload(events: EventBatchRequest): Boolean = withContext(dispatcher) {
+        Event
```
Comment (nicklasl) [inline Provider/src/main/java/com/spotify/confidence/EventSenderUploader.kt:55]: 👀

Comment (fabriziodemaria) [reply Provider/src/main/java/com/spotify/confidence/EventSenderUploader.kt:55]: Wha.... What happened there?? And how is this still compiling :D

File: Provider/src/test/java/com/spotify/confidence/EventSenderIntegrationTest.kt [hunk 2]
```diff
@@ -124,6 +124,68 @@ class EventSenderIntegrationTest {
...
+        )
+        engine.emit(
+            eventName = "my_event",
+            message = mapOf(
+                "a" to ConfidenceValue.Integer(0),
+                "message" to ConfidenceValue.Integer(1)
+            ),
+            context = mapOf(
+                "a" to ConfidenceValue.Integer(2),
+                "message" to ConfidenceValue.Integer(3)
```
Comment (nicklasl) [inline Provider/src/test/java/com/spotify/confidence/EventSenderIntegrationTest.kt:169]: I think this is going a bit overboard in this specific test.
Can we move it to a separate one titled something with `handles_message_key_collision`?

**Review by nicklasl — APPROVED** [review]

File: Provider/src/test/java/com/spotify/confidence/EventSenderIntegrationTest.kt (hunk 2 above)
Comment (fabriziodemaria) [reply Provider/src/test/java/com/spotify/confidence/EventSenderIntegrationTest.kt:169]: I renamed the test

**Review by nicklasl — APPROVED** [review]

**Review by vahidlazio — APPROVED** [review]

**Pull request merged**